### PR TITLE
Reset OOM score back to 0 for container runtime

### PR DESCRIPTION
**Pull request by mrunalp**
We don't want container runtime procesess to be unkillable
so we reset oom_score_adj back to 0 before execv
of the runtime process.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@giuseppe @haircommander ptal

**Comment by rhatdan** [comment]
LGTM

**Review by saschagrunert — APPROVED** [review]

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by giuseppe** [comment]
the container runtime exits immediately.  In what cases does it turn to be an issue?

There are some formatting issues with `make fmt`

File: src/oom.c [hunk 1]
```diff
@@ -5,16 +5,14 @@
 #include <string.h>
 #include <unistd.h>
 
-#define OOM_SCORE "-1000"
-
-void attempt_oom_adjust()
+void attempt_oom_adjust(char* oom_score)
```
Comment (haircommander) [inline src/oom.c:8]: nit can we do `const char* const` here?

Comment (haircommander) [reply src/oom.c:8]: I think there may be a line change needed in `src/oom.h` as well

Comment (mrunalp) [reply src/oom.c]: done

**Comment by kolyshkin** [comment]
> the container runtime exits immediately. In what cases does it turn to be an issue?

@giuseppe I believe this is for cases when `runc init` is waiting on the exec_fifo to continue. The only processes in cgroup are conmon and runc, both are unkillable (oom_score_adj == -1000), and the memory limit is hit, so the kernel would do an OOM kill but it can't (and there's not much memory in this cgroup to reclaim either). As a result of all this, the kernel hits code path(s) which is/are rarely hit and buggy, and bad things happen (now I am not sure what "bad things" are, perhaps this is high CPU usage by the kernel trying to find pages to reclaim, or something else entirely).

So basically this is a kludge against hitting that code path and thus those kernel bugs.

One other thing is, the whole container can inherit oom score of -1000 unless OomScoreAdj is set in its config.

**Comment by giuseppe** [comment]
> @giuseppe I believe this is for cases when `runc init` is waiting on the exec_fifo to continue. The only processes in cgroup are conmon and runc, both are unkillable (oom_score_adj == -1000), and the memory limit is hit, so the kernel would do an OOM kill but it can't (and there's not much memory in this cgroup to reclaim either). As a result of all this, the kernel hits code path(s) which is/are rarely hit and buggy, and bad things happen (now I am not sure what "bad things" are, perhaps this is high CPU usage by the kernel trying to find pages to reclaim, or something else entirely).
> 
> So basically this is a kludge against hitting that code path and thus those kernel bugs.
> 
> One other thing is, the whole container can inherit oom score of -1000 unless OomScoreAdj is set in its config.

thanks for the explanation.  That makes sense.

LGTM

**Comment by haircommander** [comment]
LGTM
